klog.go

```diff
@@ -1265,6 +1302,33 @@ func Infof(format string, args ...interface{}) {
 	logging.printf(infoLog, logging.logr, format, args...)
 }
 
+// InfoS structured logs to the INFO log.
+// The msg argument used to add constant description to the log line.
+// The key/value pairs would be join by "=" ; a newline is always appended.
+//
+// Basic examples:
+// >> klog.InfoS("Pod status updated", "pod", "kubedns", "status", "ready")
+// output:
+// >> I1025 00:15:15.525108       1 controller_utils.go:116] "Pod status updated" pod="kubedns" status="ready"
+func InfoS(msg string, keysAndValues ...interface{}) {
+	logging.printS(nil, logging.logr, msg, keysAndValues...)
+}
+
+func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
+	for i := 0; i < len(keysAndValues); i += 2 {
+		var v interface{}
+		k, ok := keysAndValues[i].(string)
+		if !ok {
+			return
+		}
+		if i+1 < len(keysAndValues) {
+			v = keysAndValues[i+1]
+			b.WriteByte(' ')
+			b.WriteString(fmt.Sprintf("%s=%q", k, v))
```

Comment:
%q isn't right for almost any inputs.

https://play.golang.org/p/gGuFvhMzxcr

Reply:
This function needs a HEFTY unit test to cover every permutation of input and the expected output.

Reply:
> %q isn't right for almost any inputs.
> 
> https://play.golang.org/p/gGuFvhMzxcr

So which one more suitable？

Reply:
> This function needs a HEFTY unit test to cover every permutation of input and the expected output.

Woking on this.

Reply:
> This function needs a HEFTY unit test to cover every permutation of input and the expected output.

working on this

Reply:
> %q isn't right for almost any inputs.
> 
> https://play.golang.org/p/gGuFvhMzxcr

how about %v?

Reply:
> This function needs a HEFTY unit test to cover every permutation of input and the expected output.

How about %#?

Reply:
> This function needs a HEFTY unit test to cover every permutation of input and the expected output.

How about %#v? https://play.golang.org/p/EaUhtFGoa_I

Reply:
> This function needs a HEFTY unit test to cover every permutation of input and the expected output.

Add some test case, please have a review?